File: PC/src/protocol/PacketType.java
package protocol;

public enum PacketType {
    TEXT((byte) 1),
    IMAGE((byte) 2),
    VIDEO((byte) 3),
    FILE((byte) 4),
    ARCHIVE((byte) 5);

    private final byte value;

    PacketType(byte value) {
        this.value = value;
    }

    public byte getValue() {
        return value;
    }

    public static PacketType fromValue(byte value) {
        for (PacketType type : values()) {
            if (type.value == value) {
                return type;
            }
        }
        return null;
    }
}

File: DocumentSend/app/src/main/java/com/example/documentsend/network/PacketType.kt
package com.example.documentsend.network

enum class PacketType(val value: Byte) {
    TEXT(1),
    IMAGE(2),
    VIDEO(3),
    FILE(4),
    ARCHIVE(5);

    companion object {
        fun fromValue(value: Byte): PacketType? {
            return values().find { it.value == value }
        }
    }
}

